src/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor_Helpers.cs

```diff
@@ -146,7 +146,24 @@ static string GetLanguageSpecificId(string? language, string noLanguageId, strin
             // Create driver that holds onto compilation and associated analyzers
             var filteredProjectAnalyzers = projectAnalyzers.WhereAsArray(static a => !a.IsWorkspaceDiagnosticAnalyzer());
             var filteredHostAnalyzers = hostAnalyzers.WhereAsArray(static a => !a.IsWorkspaceDiagnosticAnalyzer());
-            var filteredProjectSuppressors = filteredProjectAnalyzers.WhereAsArray(static a => a is DiagnosticSuppressor);
+            var hostAnalyzersSegmentWithSuppressors = filteredHostAnalyzers.AsMemory()[filteredHostAnalyzers.FindRange(static a => a is DiagnosticSuppressor)];
+
+            // Add any project suppressors which are not already included in filteredHostAnalyzers
+            var filteredProjectSuppressors = filteredProjectAnalyzers.WhereAsArray(
+                static (a, hostAnalyzersSegmentWithSuppressors) =>
+                {
+                    if (a is not DiagnosticSuppressor)
+                        return false;
+
+                    foreach (var existing in hostAnalyzersSegmentWithSuppressors.Span)
+                    {
+                        if (Equals(existing, a))
+                            return false;
+                    }
```

Comment:
what's this check for?  can this just be: `a is DiagSuppressor && !hostAnalyzersSegmentWithSuppressors.Contains(a)`?

Reply:
Ideally, yes. In practice, one of our TFMs doesn't define `Memory<T>.Contains`.

Reply:
➡️ Switched to different approach